lib/browser_collection.js

```diff
@@ -46,7 +46,24 @@ class BrowserCollection {
     return this.browsers.map((browser) => browser.serialize())
   }
 
-  getResults () {
+  calculateExitCode (results, singleRunBrowserNotCaptured, failOnEmptyTestSuite, failOnFailingTestSuite) {
+    if (results.disconnected || singleRunBrowserNotCaptured) {
+      return 1
+    }
+    if (results.success + results.failed === 0 && !failOnEmptyTestSuite) {
+      return 0
+    }
+    if (results.error) {
```

Comment:
I believe that at least some user errors (syntax?) come out as results.errors rather than result.failed.  But it's your feature to decide, or maybe try it and see.

Reply:
Yes, syntax errors will set `results.error` to true. But from my point of view that is an other thing than a failure and shall not lead to a successful exit code
The syntax error will not generate a test failure in the generated test report, and may then be missed if hidden by the `failOnEmptyTestSuite` flag